### PR TITLE
Update Beam SDK dependency page and linked material.

### PR DESCRIPTION
**Pull request by tvalentyn**

### CODE REVIEW

**Comment by tvalentyn** [comment]
R: @rszper

**Comment by tvalentyn** [comment]
TODO: check staged website before merging.

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 1]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
 
 # Beam SDK for Python dependencies
 
-The Beam SDKs depend on common third-party components which then
-import additional dependencies. Version collisions can result in unexpected
-behavior in the service. If you are using any of these packages in your code, be
-aware that some libraries are not forward-compatible and you may need to pin to
-the listed versions that will be in scope during execution.
+This page provides the information on dependencies of Apache Beam Python SDK
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:21]:
```suggestion
This page provides the information about the Apache Beam Python SDK dependencies.
```

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 2]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
...
+itself.
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:22]:
```suggestion
```

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 3]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
...
 
-Dependencies for your Beam SDK version are listed in `setup.py` in the Beam repository. To view them, perform the following steps:
+If your pipeline requires addtional dependencies, see the [Managing Python Pipeline Dependencies](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/) page.
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:24]:
```suggestion
If your pipeline requires additional dependencies, see [Managing Python Pipeline Dependencies](https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/).
```

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 4]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
...
+
+Dependencies of the Apache Beam Python SDK are defined in the `setup.py` file in the Beam repository. To view them, perform the following steps:
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:26]:
```suggestion
Dependencies of the Apache Beam Python SDK are defined in the `setup.py` file in the Beam repository. To view them, take the following steps:
```

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 5]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
...
 
 1. Open `setup.py`.
 
     ```
-    https://raw.githubusercontent.com/apache/beam/v<VERSION_NUMBER>/sdks/python/setup.py
+    https://github.com/apache/beam/blob/release-<VERSION_NUMBER>/sdks/python/setup.py
     ```
 
-    <p class="paragraph-wrap">Replace `&lt;VERSION_NUMBER&gt;` with the major.minor.patch version of the SDK. For example, <a href="https://raw.githubusercontent.com/apache/beam/v{{< param release_latest >}}/sdks/python/setup.py" target="_blank" rel="noopener noreferrer">https://raw.githubusercontent.com/apache/beam/v{{< param release_latest >}}/sdks/python/setup.py</a> will provide the dependencies for the {{< param release_latest >}} release.</p>
+    <p class="paragraph-wrap">Replace `&lt;VERSION_NUMBER&gt;` with the major.minor.patch version of the SDK. For example, <a href="https://github.com/apache/beam/blob/release-{{< param release_latest >}}/sdks/python/setup.py" target="_blank" rel="noopener noreferrer">https://github.com/apache/beam/blob/release-{{< param release_latest >}}/sdks/python/setup.py</a> will provide the dependencies for the {{< param release_latest >}} release.</p>
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:34]:
```suggestion
    <p class="paragraph-wrap">Replace `&lt;VERSION_NUMBER&gt;` with the major.minor.patch version of the SDK. For example, <a href="https://github.com/apache/beam/blob/release-{{< param release_latest >}}/sdks/python/setup.py" target="_blank" rel="noopener noreferrer">https://github.com/apache/beam/blob/release-{{< param release_latest >}}/sdks/python/setup.py</a> provides the dependencies for the {{< param release_latest >}} release.</p>
```

File: website/www/site/content/en/documentation/sdks/python-dependencies.md [hunk 6]
```diff
@@ -18,26 +18,25 @@ limitations under the License.
...
 
 
 2. Review the core dependency list under `REQUIRED_PACKAGES`.
 
-    **Note:** If you require [extra features](/get-started/quickstart-py#extra-requirements) such as `gcp` or `test`, you should review the lists under `REQUIRED_TEST_PACKAGES`, `GCP_REQUIREMENTS`, or `INTERACTIVE_BEAM` for additional dependencies.
+    **Note:** If you require [extra features](/get-started/quickstart-py#extra-requirements) such as `gcp` or `dataframe`, you should review the lists under `extras_require` for additional dependencies.
```
Comment (rszper) [inline website/www/site/content/en/documentation/sdks/python-dependencies.md:39]:
```suggestion
    **Note:** If you need [extra features](/get-started/quickstart-py#extra-requirements), such as `gcp` or `dataframe`, review the lists in `extras_require` for additional dependencies.
```

**Review by rszper — APPROVED** [review]

**Pull request merged**